src/Microsoft.Extensions.Primitives/CompositeChangeToken.cs

```diff
@@ -0,0 +1,126 @@
+        private bool _registeredCallbackProxy;
+        private bool _canBeChanged = true;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="CompositeChangeToken"/>.
+        /// </summary>
+        /// <param name="changeTokens">The list of <see cref="IChangeToken"/> to compose.</param>
+        public CompositeChangeToken(IReadOnlyList<IChangeToken> changeTokens)
+        {
+            if (changeTokens == null)
+            {
+                throw new ArgumentNullException(nameof(changeTokens));
+            }
+
+            ChangeTokens = changeTokens;
+        }
+
+        /// <summary>
+        /// Returns the list of <see cref="IChangeToken"/> which compose the current <see cref="CompositeChangeToken"/>.
+        /// </summary>
+        public IReadOnlyList<IChangeToken> ChangeTokens { get; }
+
+        /// <inheritdoc />
+        public virtual IDisposable RegisterChangeCallback(Action<object> callback, object state)
+        {
+            if (!_canBeChanged)
+            {
+                return NullDisposable.Singleton;
+            }
+
+            EnsureCallbacksRegistered();
+            return _cancellationTokenSource.Token.Register(callback, state);
+        }
+
+        /// <inheritdoc />
+        public bool HasChanged
+        {
+            get
+            {
+                for (var i = 0; i < ChangeTokens.Count; i++)
+                {
+                    if (ChangeTokens[i].HasChanged)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        /// <inheritdoc />
+        public bool ActiveChangeCallbacks
+        {
+            get
+            {
+                for (var i = 0; i < ChangeTokens.Count; i++)
+                {
+                    if (ChangeTokens[i].ActiveChangeCallbacks)
+                    {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+        }
+
+        private void EnsureCallbacksRegistered()
+        {
+            if (_registeredCallbackProxy)
+            {
+                return;
+            }
+
+            for (var i = 0; i < ChangeTokens.Count; i++)
+            {
+                if (ActiveChangeCallbacks)
+                {
+                    ChangeTokens[i].RegisterChangeCallback(_onChangeDelegate, this);
+                }
+            }
+            _registeredCallbackProxy = true;
+        }
+
+        private static void OnChange(object state)
+        {
+            var compositeChangeTokenState = (CompositeChangeToken)state;
+            if (!compositeChangeTokenState._canBeChanged)
+            {
+                return;
+            }
+
+            compositeChangeTokenState._canBeChanged = false;
+            compositeChangeTokenState._cancellationTokenSource.Cancel();
```

Comment:
Dispose the IDisposable returned from `RegisterChangeCallback` for each token? ChangeTokens don't implement IDisposable.